capsules/core/src/lib.rs

```diff
@@ -3,6 +3,12 @@
 // Copyright Tock Contributors 2023.
 
 #![forbid(unsafe_code)]
+#![cfg_attr(
+    target_feature = "xcheri",
+    feature(result_option_inspect),
+    feature(int_roundings),
+    feature(is_some_and)
+)]
```

Comment:
I think we need a boilerplate comment for all of these briefly explaining why they are there and maybe a pointer to more info.

Reply:
I have added a comment to all the these CHERI specific cfgs